test/lit/parse-error-func-param-type.wast

```diff
@@ -0,0 +1,11 @@
+;; This function's type does not match the param we define for it.
+
+;; RUN: not wasm-opt %s 2>&1 | filecheck %s
+;; CHECK: Fatal: 5:10: error: type does not match provided signature
+
+(module
+ (type $0 (func))
+
+ (func $0 (type $0) (param $var$0 i32)
+ )
```

Comment:
Might as well move the dangling paren up if we're not running the update script.
```suggestion
 (func $0 (type $0) (param $var$0 i32))
```